### PR TITLE
Add BaseExceptionGroup

### PR DESCRIPTION
**Pull request by abyesilyurt**
Add BaseExceptionGroup that is introduced in Python 3.11. ExceptionGroup still needs to be implemented.

#4628 

[1]: https://docs.python.org/3/library/exceptions.html#exception-groups

### CODE REVIEW

File: Lib/test/pickletester.py [hunk 1]
```diff
@@ -765,6 +765,7 @@ def _reconstruct(cls, obj, kwargs):
     FloatingPointError,
     FutureWarning,
     GeneratorExit,
+    BaseExceptionGroup,
```
Comment (youknowone) [inline Lib/test/pickletester.py:768]:
```suggestion
```

**Review by youknowone — CHANGES_REQUESTED** [review]
looks good, let's see how test goes without the line

**Comment by youknowone** [comment]
hmm. updating pickle/test_pickle by #4564 will be helpful

**Comment by abyesilyurt** [comment]
Updated test_pickle.py, but commented out `ExceptionGroup`, as it still needs to be implemented. I will leave for the next PR.

**Comment by fanninpm** [comment]
The `update test_pickle.py` commit should mention the version of CPython you got the updates from.

File: Lib/test/test_pickle.py [hunk 2]
```diff
@@ -632,7 +632,9 @@ def test_exceptions(self):
                            ResourceWarning,
                            StopAsyncIteration,
                            RecursionError,
-                           EncodingWarning):
+                           EncodingWarning,
+                           #ExceptionGroup,
```
[the remark this answers is not in the document]
Comment (abyesilyurt) [reply Lib/test/test_pickle.py]: added

**Comment by abyesilyurt** [comment]
Thanks, added CPython version to the commit message.

File: Lib/test/test_pickle.py [hunk 3]
```diff
@@ -632,7 +632,9 @@ def test_exceptions(self):
                            ResourceWarning,
                            StopAsyncIteration,
                            RecursionError,
-                           EncodingWarning):
+                           EncodingWarning,
+                           #ExceptionGroup, # TODO: RUSTPYTHON
```
Comment (fanninpm) [inline Lib/test/test_pickle.py:636]: I'd also like for the _edits_ to this particular line to live in a separate commit.

Comment (abyesilyurt) [reply Lib/test/test_pickle.py:636]: alright :D done

File: Lib/test/test_pickle.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (abyesilyurt) [reply Lib/test/test_pickle.py]: I see what you mean, it should be good this time.

**Review by youknowone — APPROVED** [review]
looks good, thank you for contributing!

**Pull request merged**